pandas/tests/interchange/test_impl.py

```diff
@@ -143,9 +127,25 @@ def test_bitmasks_pyarrow(offset, length, expected_values):
 
 
 @pytest.mark.parametrize(
-    "data", [int_data, uint_data, float_data, bool_data, datetime_data]
+    "data",
+    [
+        lambda: random.randint(-100, 100),
```

Comment:
Just want to be sure - I think the use of the lambda ensures that test collection is deterministic, is that right?

Reply:
The lambda here just makes the data generation lazy and not evaluated during the test collection phase, so the data isn't necessarily deterministic.